internal/allocator/allocator.go

```diff
@@ -291,6 +293,10 @@ func (a *Allocator) AllocateFromPool(svcKey string, svc *v1.Service, serviceIPFa
 		}
 	}
 
+	if len(ipfamilySel) == 1 && serviceIPFamily == ipfamily.PreferDualStack {
```

Comment:
this just takes the first iiuc. We need a more refined logic:

In case of a dual stack service, what metallb should do is:

    look for dual stack pools
    If not available, look for single stack pools for the primary family
    If not available, look for single stack pools for the secondary family

Reply:
this, with the addition of the fact that we should match what's filled for a clusterip imo